lib/foreman_maintain.rb

```diff
@@ -176,18 +176,32 @@ def pkg_and_cmd_name
 
     def perform_self_upgrade
       package_name, command = pkg_and_cmd_name
+      package_manager = ForemanMaintain.package_manager
 
       puts "Checking for new version of #{package_name}..."
-      if ForemanMaintain.package_manager.update_available?(main_package_name)
+
+      enable_maintenance_module
+
+      if package_manager.update_available?(main_package_name)
         puts "\nUpdating #{package_name} package."
-        ForemanMaintain.package_manager.update(main_package_name, :assumeyes => true)
+        package_manager.update(main_package_name, :assumeyes => true)
         puts "\nThe #{package_name} package successfully updated."\
              "\nRe-run #{command} with required options!"
         exit 75
       end
       puts "Nothing to update, can't find new version of #{package_name}."
     end
 
+    def enable_maintenance_module
+      maintenance_module = 'satellite-maintenance:el8'
+
+      if ForemanMaintain.package_manager.module_exists?(maintenance_module) &&
+         !package_manager.module_enabled?(maintenance_module)
+        puts "\nEnabling #{maintenance_module} module"
+        ForemanMaintain.package_manager.enable_module(maintenance_module)
+      end
```

Comment:
`package_manager` is not defined inside this method, so it might use the one from the outer scope, or might crash.